src/relaycard/card.py

```diff
@@ -47,70 +56,70 @@ def execute(self, command, address=0, data=0):
         return self.send_frame(RequestFrame(command, address, data))
 
     def execute_retry(self, command, validator, address=0, data=0, retries=3):
-        for i in range(1, retries+1):
+        for i in range(1, retries + 1):
             try:
                 response = self.execute(command, address, data)
                 assert validator(response)
                 return response
-            except AssertionError, e:
-                logging.error('Error, retry #%d: %s' % (i, e))
+            except AssertionError as e:
```

Comment:
This is quite old code and not best practice (which I only learned over the years). If you are motivated we should get rid of all the asserts as they are not meant to be used in production code. we might simply introduce a `RelayCardException` to make it not to overly complicated but not using `assert` is a think to consider. (see https://snyk.io/blog/the-dangers-of-assert-in-python/ for more details)

Reply:
Done. Have a look. I think most of the exceptions can be removed, when creating a proper types with ranges https://docs.python.org/3/library/typing.html#typing.Annotated